src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -49,6 +56,40 @@ public class XdocsTemplateSink extends XdocSink {
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
         this.encoding = encoding;
+        this.writer = new PrintWriter(writer);
+    }
+
+    /**
+     * Customization of org.apache.maven.doxia.sink.impl.XhtmlBaseSink.unifyEOL() method.
+     *
+     * @param text the text to scan.
+     * @return a String that contains only Linux System EOLs.
+     */
+    protected String unifyToLinuxEndOfLine(String text) {
+        return text
+                .replace("\r\n", Character.toString(XdocsTemplateSink.EOL))
+                .replace('\r', XdocsTemplateSink.EOL);
+    }
+
+    /**
+     * Overriden from org.apache.maven.doxia.sink.impl.AbstractXmlSink class.
+     *
+     * @param text the text to write to the destination.
+     */
+    @Override
+    protected void write(String text) {
+        if (text != null) {
+            writer.write(unifyToLinuxEndOfLine(text));
+        }
+    }
+
+    /**
+     * Overriden from org.apache.maven.doxia.sink.impl.AbstractXmlSink class.
+     * Writes a Linux System endOfLine.
+     */
+    @Override
+    protected void writeEOL() {
+        write(Character.toString(this.EOL));
```

Comment:
why it is not enough to override only this method
https://github.com/apache/maven-doxia/blob/240f2b7ffe00c29446f6f6006327ff9394e02043/doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractXmlSink.java#L142-L144

why we need all others ?

Reply:
> why we need all others ?

We need to override the `write()` method, otherwise the implementation would be directed to this.

https://github.com/apache/maven-doxia/blob/240f2b7ffe00c29446f6f6006327ff9394e02043/doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java#L2006-L2014

And then the control flow will pass through the `unifyEOL` method and all our efforts will go in vain.

Reply:
please reference in javadoc of your method that you override Xhtml5BaseSink method.

Why skipping `if (!this.tableCaptionXMLWriterStack.isEmpty() && this.tableCaptionXMLWriterStack.getLast() != null) {` is ok ?

Reply:
@kalpadiptyaroy , ping.

Reply:
I skipped them because they were redundant in our context. Do you want me to include them?

Reply:
If we copy code, I would copy it completely and has change only in line end , to avoid unexpected problems.

At least, next person who deals with this method knows what method we override from what class and we changed only eol.

We can skip extra code that is not required for now for us, but in this case we need to explain this in javadoc.

Reply:
Sure! I will update java docs